src/cleanup/security-group-cleaner.ts

```diff
@@ -79,3 +87,52 @@ function arrayContains(arr: string[], set: Set<string>): boolean {
 function filterOut(arr: string[], set: Set<string>): string[] {
   return arr.filter(el => !set.has(el));
 }
+export async function CleanupElasticacheSecurityGroups(
+  groupIds: Set<string>
+): Promise<void> {
+  const CacheClusters = await getRawCacheClusters();
+  const ReplicationGroups = await getRawReplicationGroups();
+
+  for (const ReplicationGroup of ReplicationGroups) {
+    await cleanReplicationGroup(ReplicationGroup, CacheClusters, groupIds);
+  }
+}
+
+async function cleanReplicationGroup(
+  replicationGroup: ReplicationGroup,
+  CacheClusters: CacheCluster[],
+  groupIds: Set<string>
+): Promise<void> {
+  if (
+    replicationGroup.NodeGroups === undefined ||
+    replicationGroup.NodeGroups[0] === undefined ||
+    replicationGroup.NodeGroups[0].NodeGroupMembers === undefined ||
+    replicationGroup.NodeGroups[0].NodeGroupMembers[0] === undefined
+  )
+    return;
+  const exampleCacheCluster =
+    replicationGroup.NodeGroups[0].NodeGroupMembers[0].CacheClusterId;
+  const cacheSecurityGroups = CacheClusters.find(
+    cache => cache.CacheClusterId === exampleCacheCluster
+  );
+  if (
+    cacheSecurityGroups === undefined ||
+    cacheSecurityGroups.SecurityGroups === undefined
+  )
+    return;
+  const cacheSecurityGroupsIds: string[] = [
+    ...cacheSecurityGroups.SecurityGroups.flatMap(
+      group => group.SecurityGroupId ?? ' '
+    ).filter(id => id !== ' ' && groupIds.has(id)),
+  ];
+
+  if (
+    cacheSecurityGroupsIds.length !== cacheSecurityGroups.SecurityGroups.length
+  ) {
+    await modifyElasticacheReplicationGroup({
+      identifier: replicationGroup.ReplicationGroupId,
+      securityGroupIds: cacheSecurityGroupsIds,
+      cachePreviousSecurityGroups: cacheSecurityGroups.SecurityGroups,
```

Comment:
Could you please, briefly explain why do we need this third argument? In which cases can there be 0 `securityGroupIds`?